R/NBth.R

```diff
@@ -277,69 +277,69 @@ setMethod(
-
-        rownames(para) <- c("signal", "r")
+  "fitNBth", "matrix",
+  function(object, features_high, probenum, sizefact_BG, sizefact_start = sizefact_BG, size_scale = c("sum", "first"), threshold_start, threshold_fix = FALSE, tol = 1e-7, iterations = 8,
+           start_para = c(threshold_start, 1), lower_sizefact = 0, lower_threshold = threshold_start / 5) {
+    size_scale <- match.arg(size_scale)
+    sizefact0 <- sizefact <- sizefact_start
+    threshold <- threshold_start
+    # mat <- matrix(1, nrow(object), 1)
+    if (is.null(names(probenum))) names(probenum) <- rownames(object)
+    for (iter in seq_len(iterations)) {
+      para <- NBth_paraopt(object[features_high, ], probenum[features_high], sizefact, sizefact_BG, threshold, start = start_para)
+      # result <- mleprobeNBall(object[,features_high], mat, sizefact0, sizefact,
+      #                         matrix(0,1,1), threshold, 0,
+      #                         c(rep(0,ncol(mat)), 1, threshold), 0)
+      features_NA <- features_high[unique(which(is.na(para), arr.ind = TRUE)[, 2])]
+      features_remain <- setdiff(features_high, features_NA)
+      
+      
+      for (i in seq_len(length(sizefact))) {
+        fun <- NBth_scalenll(object[features_remain, i], probenum[features_remain], t(para[1, features_remain]), t(para[2, features_remain]), sizefact_BG[i], threshold)
+        sizefact[i] <- optim(c(sizefact_start[i]), fun, lower = c(lower_sizefact), method = "L-BFGS-B")$par
+      }
+      
+      if (size_scale == "first") {
+        scale_fac <- sizefact[1]
+      } else if (size_scale == "sum") {
+        scale_fac <- sum(sizefact)
+      }
+      
+      sizefact <- sizefact / scale_fac
+      
+      
+      if (!threshold_fix) {
+        fun1 <- NBth_thnll(object[features_remain, ], probenum[features_remain], sizefact, sizefact_BG, scale_fac * t(para[1, features_remain]), t(para[2, features_remain]))
         
-        return(list(
-            para0 = NA,
-            para = para,
-            sizefact = sizefact,
-            preci1 = NA,
-            conv0 = NA,
-            conv = NA,
-            Im = NA,
-            features_high = features_high,
-            features_all = NA,
-            threshold = threshold
-        ))
+        threshold <- optim(c(threshold_start), fun1, lower = c(lower_threshold), method = "L-BFGS-B")$par
+      }
+      
+      
+      
+      
+      message(sprintf("Iteration = %s, squared error = %s", iter, sum((sizefact - sizefact0)^2)))
+      
+      
+      if (sum((sizefact - sizefact0)^2) < tol) break
+      
+      sizefact0 <- sizefact
+      
     }
-)
+    message("Model converged.")
+    
+    rownames(para) <- c("signal", "r")
+    
+    return(list(
+      para0 = NA,
+      para = para,
+      sizefact = sizefact,
+      preci1 = NA,
+      conv0 = NA,
+      conv = NA,
+      Im = NA,
+      features_high = features_high,
+      features_all = NA,
+      threshold = threshold
+    ))
+  }
+)
```

Comment:
Add new line. Also, it looks like nothing was changed in this file, but spacing. Is there anyway to remove your changes such as these so we don't have to double-check everything without any changes?

Reply:
I don't think anything was changed. I am not sure how to roll this one file back.

Reply:
Is it only this file? It looks like a lot of files were "changed", but not really. Do you know which these are so I can skip?

Reply:
@ned-procogia can you change your IDE spacing to default to 4 instead of 2 spaces and commit. This is a requirement for our bioconductor packages so should not be changed, and this has made it extremely difficult to code review with all the supposed changes showing.